### ISSUE TITLE
trim-filenames no longer works

### ISSUE THREAD

**Issue by millette** [yt-dlp/yt-dlp#11016]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Provide a description that is worded well enough to be understood

The --trim-filenames option no longer works

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', '--trim-filenames', '30', 'https://www.facebook.com/share/r/MshCef9v9W6dGJwU/?mibextid=ICIPch']
[debug] User config "/home/millette/.config/yt-dlp/config": ['--trim-filenames', '199']
[debug] Encodings: locale UTF-8, fs utf-8, pref UTF-8, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version nightly@2024.09.14.232748 from yt-dlp/yt-dlp-nightly-builds (linux_exe)
[debug] Python 3.11.10 (CPython x86_64 64bit) - Linux-6.9.3-76060903-generic-x86_64-with (OpenSSL 3.1.7 3 Sep 2024)
[debug] exe versions: ffmpeg 4.4.2 (setts), ffprobe 4.4.2, phantomjs broken, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.08.30, curl_cffi-0.7.1, mutagen-1.47.0, requests-2.32.3, secretstorage-3.3.3, sqlite3-3.44.2, urllib3-2.2.3, websockets-13.0.1
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Loaded 1835 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-nightly-builds/releases/latest
Latest version: nightly@2024.09.14.232748 from yt-dlp/yt-dlp-nightly-builds
yt-dlp is up to date (nightly@2024.09.14.232748 from yt-dlp/yt-dlp-nightly-builds)
[generic] Extracting URL: https://www.facebook.com/share/r/MshCef9v9W6dGJwU/?mibextid=ICIPch
[generic] ?mibextid=ICIPch: Downloading webpage
[redirect] Following redirect to https://www.facebook.com/reel/3783327605240707?mibextid=ICIPch&rdid=AlsCFt7wwjwiqEsS&share_url=https%3A%2F%2Fwww.facebook.com%2Fshare%2Fr%2FMshCef9v9W6dGJwU%2F%3Fmibextid%3DICIPch
[facebook:reel] Extracting URL: https://www.facebook.com/reel/3783327605240707?mibextid=ICIPch&rdid=AlsCFt7wwjwiqEsS&share_url=https%3A%2F%2Fwww.facebook.com%2Fshare%2Fr%2FMshCef9v9W6dGJwU%2F%3Fmibextid%3DICIPch
[facebook] Extracting URL: https://m.facebook.com/watch/?v=3783327605240707&_rdr
[facebook] 3783327605240707: Downloading webpage
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), channels, acodec, size, br, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] 3783327605240707: Downloading 1 format(s): 1234247767607610v+798025075604220a
[debug] Invoking http downloader on "https://video.fymq1-2.fna.fbcdn.net/v/t39.25447-2/459184198_879162244126649_953794393263884872_n.mp4?_nc_cat=1&ccb=1-7&_nc_sid=9a5d50&efg=eyJ2ZW5jb2RlX3RhZyI6ImRhc2hfcjJhdjEtcjFnZW4ydnA5X3E5MCIsInZpZGVvX2lkIjozNzgzMzI3NjA1MjQwNzA3fQ==&_nc_ohc=Qm1m3okIDX0Q7kNvgGsxsRM&_nc_ht=video.fymq1-2.fna&_nc_gid=AiwL4M-t4XSQrS2MFrcGUqt&oh=00_AYAorCE5hfG4wm9sOzp4ZwXmUEjq5MdVMxILn99F33Jk5Q&oe=66EE4DF5"
ERROR: unable to open for writing: [Errno 36] Filename too long: '672K views · 36K reactions ｜ F. #funny #funnyvideo #funnyreels #fyp #humor #trending #reels #foryoupage\xa0\xa0 #foryou #funnyvideos #viral #viralreels\xa0 #video #fun #videos #funnypages #crazy\xa0 #goviral #laughing #dailyvideo #reel\xa0 #funnymoments #funnyclips #fail\xa0 #haha #hilarious #lol #usa #fails ｜ Funny memes & videos ｜ funny_dailymoments · Original audio [3783327605240707].f1234247767607610v.mp4.part'
Traceback (most recent call last):
  File "yt_dlp/utils/_utils.py", line 591, in sanitize_open
  File "yt_dlp/utils/_utils.py", line 1593, in __init__
OSError: [Errno 36] Filename too long: '672K views · 36K reactions ｜ F. #funny #funnyvideo #funnyreels #fyp #humor #trending #reels #foryoupage\xa0\xa0 #foryou #funnyvideos #viral #viralreels\xa0 #video #fun #videos #funnypages #crazy\xa0 #goviral #laughing #dailyvideo #reel\xa0 #funnymoments #funnyclips #fail\xa0 #haha #hilarious #lol #usa #fails ｜ Funny memes & videos ｜ funny_dailymoments · Original audio [3783327605240707].f1234247767607610v.mp4.part'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "yt_dlp/downloader/http.py", line 265, in download
  File "yt_dlp/downloader/common.py", line 250, in wrapper
  File "yt_dlp/downloader/common.py", line 236, in error_callback
  File "yt_dlp/utils/_utils.py", line 5215, in report_retry
  File "yt_dlp/downloader/common.py", line 245, in wrapper
  File "yt_dlp/downloader/common.py", line 256, in sanitize_open
  File "yt_dlp/utils/_utils.py", line 593, in sanitize_open
OSError: [Errno 36] Filename too long: '672K views · 36K reactions ｜ F. #funny #funnyvideo #funnyreels #fyp #humor #trending #reels #foryoupage\xa0\xa0 #foryou #funnyvideos #viral #viralreels\xa0 #video #fun #videos #funnypages #crazy\xa0 #goviral #laughing #dailyvideo #reel\xa0 #funnymoments #funnyclips #fail\xa0 #haha #hilarious #lol #usa #fails ｜ Funny memes & videos ｜ funny_dailymoments · Original audio [3783327605240707].f1234247767607610v.mp4.part'


[debug] Invoking http downloader on "https://video.fymq1-2.fna.fbcdn.net/v/t39.25447-2/459043564_3668173456733131_3890287003599920769_n.mp4?_nc_cat=104&ccb=1-7&_nc_sid=9a5d50&efg=eyJ2ZW5jb2RlX3RhZyI6ImRhc2hfYXVkaW9fYWFjcF82NF9mbm9ybTE0X2ZyYWdfMl9hdWRpbyIsInZpZGVvX2lkIjozNzgzMzI3NjA1MjQwNzA3fQ==&_nc_ohc=HDYydHwZp_oQ7kNvgH-B08c&_nc_ht=video.fymq1-2.fna&_nc_gid=AiwL4M-t4XSQrS2MFrcGUqt&oh=00_AYDREBTIwyJFQ8vMy6ER9yolc9eFH9tClpGOQyLbDKXL2Q&oe=66EE3BD7"
ERROR: unable to open for writing: [Errno 36] Filename too long: '672K views · 36K reactions ｜ F. #funny #funnyvideo #funnyreels #fyp #humor #trending #reels #foryoupage\xa0\xa0 #foryou #funnyvideos #viral #viralreels\xa0 #video #fun #videos #funnypages #crazy\xa0 #goviral #laughing #dailyvideo #reel\xa0 #funnymoments #funnyclips #fail\xa0 #haha #hilarious #lol #usa #fails ｜ Funny memes & videos ｜ funny_dailymoments · Original audio [3783327605240707].f798025075604220a.m4a.part'
Traceback (most recent call last):
  File "yt_dlp/utils/_utils.py", line 591, in sanitize_open
  File "yt_dlp/utils/_utils.py", line 1593, in __init__
OSError: [Errno 36] Filename too long: '672K views · 36K reactions ｜ F. #funny #funnyvideo #funnyreels #fyp #humor #trending #reels #foryoupage\xa0\xa0 #foryou #funnyvideos #viral #viralreels\xa0 #video #fun #videos #funnypages #crazy\xa0 #goviral #laughing #dailyvideo #reel\xa0 #funnymoments #funnyclips #fail\xa0 #haha #hilarious #lol #usa #fails ｜ Funny memes & videos ｜ funny_dailymoments · Original audio [3783327605240707].f798025075604220a.m4a.part'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "yt_dlp/downloader/http.py", line 265, in download
  File "yt_dlp/downloader/common.py", line 250, in wrapper
  File "yt_dlp/downloader/common.py", line 236, in error_callback
  File "yt_dlp/utils/_utils.py", line 5215, in report_retry
  File "yt_dlp/downloader/common.py", line 245, in wrapper
  File "yt_dlp/downloader/common.py", line 256, in sanitize_open
  File "yt_dlp/utils/_utils.py", line 593, in sanitize_open
OSError: [Errno 36] Filename too long: '672K views · 36K reactions ｜ F. #funny #funnyvideo #funnyreels #fyp #humor #trending #reels #foryoupage\xa0\xa0 #foryou #funnyvideos #viral #viralreels\xa0 #video #fun #videos #funnypages #crazy\xa0 #goviral #laughing #dailyvideo #reel\xa0 #funnymoments #funnyclips #fail\xa0 #haha #hilarious #lol #usa #fails ｜ Funny memes & videos ｜ funny_dailymoments · Original audio [3783327605240707].f798025075604220a.m4a.part'
```

**Comment by Grub4K** [comment]
Duplicate of #2314 or #3494, whichever you prefer.

TLDR: `--trim-filenames` is badly designed and doesn't always work as expected and you should use something like `-o "%(title).30B [%(id)s].%(ext)s"` to trim it instead. See above issues for reasoning and discussion

**Comment by millette** [comment]
@Grub4K Thanks, I saw those issues but thought they were more specific, since I the option was working fine for me before. I'll use -o instead from now on.
